File: react-workspace/todo-ui/src/services/TodoService.js
import axios from "axios";

const BASE_REST_API_URL='http://localhost:8080/api/todos';

export const getAllTodos=()=>{
    return axios.get(BASE_REST_API_URL);
}

export const saveTodo=(todo)=>{
    return axios.post(BASE_REST_API_URL,todo);
}

export const getTodo=(id)=>{
    return axios.get(BASE_REST_API_URL + '/' + id);
}

export const updateTodo=(id,todo)=>{
    return axios.put(BASE_REST_API_URL + '/' + id,todo);
}

export const deleteTodo=(id)=>{
    return axios.delete(BASE_REST_API_URL + '/' + id);
}

export const completeTodo=(id)=>{
    return axios.patch(BASE_REST_API_URL + '/' + id + '/complete');
}

export const inCompleteTodo=(id)=>{
    return axios.patch(BASE_REST_API_URL + '/' + id + '/in-complete');
}

File: todo-management/src/main/java/net/project/todo/service/TodoService.java
package net.project.todo.service;

import net.project.todo.dto.TodoDto;

import java.util.List;

public interface TodoService {
    TodoDto addTodo(TodoDto todoDto);

    TodoDto getTodo(Long id);

    List<TodoDto> getAllTodos();

    TodoDto updateTodo(TodoDto todoDto,Long id);

    void deleteTodo(Long id);

    TodoDto completeTodo(Long id);

    TodoDto inCompleteTodo(Long id);








}
